adapters/Snap/SnapAdapter/GADMediationAdapterSnap.m

```diff
@@ -14,71 +14,115 @@
 
 #import "GADMediationAdapterSnap.h"
 
+#import "GADMediationSnapBanner.h"
+#import "GADMediationSnapRewardedAd.h"
+#import "GADMediationSnapInterstitial.h"
 #import "GADMediationAdapterSnapConstants.h"
 
-@implementation GADMediationAdapterSnap
+#import <SAKSDK/SAKSDK.h>
 
-- (void)collectSignalsForRequestParameters:(nonnull GADRTBRequestParameters *)params
-                         completionHandler:(nonnull GADRTBSignalCompletionHandler)completionHandler {
-  // TODO: Start signal generation and call completionHandler when the signal generation finishes or fails.
-  completionHandler("", nil);
+@implementation GADMediationAdapterSnap {
+    GADMediationSnapBanner *_banner;
+    GADMediationSnapInterstitial *_interstitial;
+    GADMediationSnapRewardedAd *_rewardedAd;
 }
 
-+ (void)setUpWithConfiguration:(nonnull GADMediationServerConfiguration *)configuration
-             completionHandler:(nonnull GADMediationAdapterSetUpCompletionBlock)completionHandler {
-  // TODO: Set up Snap ad network SDK and perform any necessary prefetching or configuration work.
-  completionHandler(nil);
++ (GADVersionNumber)adapterVersion
+{
+    NSArray<NSString *> *versionComponents = [GADMAdapterSnapVersion componentsSeparatedByString:@"."];
+    GADVersionNumber version = {0};
+    if (versionComponents.count == 3) {
+      version.majorVersion = versionComponents[0].integerValue;
+      version.minorVersion = versionComponents[1].integerValue;
+      version.patchVersion = versionComponents[2].integerValue;
+    }
+    return version;
+}
+
++ (GADVersionNumber)adSDKVersion
+{
+    NSArray<NSString *> *versionComponents = [SAKMobileAd.shared.sdkVersion componentsSeparatedByString:@"."];
+    GADVersionNumber version = {0};
+    if (versionComponents.count == 3) {
+      version.majorVersion = versionComponents[0].integerValue;
+      version.minorVersion = versionComponents[1].integerValue;
+      version.patchVersion = versionComponents[2].integerValue;
+    }
+    return version;
 }
 
-+ (GADVersionNumber)adSDKVersion {
-  // TODO: Populate `versionString` with the ad-network's SDK version in NSString format.
-  NSString *versionString = @"";
-  GADVersionNumber version = {0};
-  NSArray<NSString *> *components = [versionString componentsSeparatedByString:@"."];
-  if (components.count == 3) {
-    version.majorVersion = components[0].integerValue;
-    version.minorVersion = components[1].integerValue;
-    version.patchVersion = components[2].integerValue;
-  } else {
-    NSLog(@"Unexpected ad SDK version string: %@. Returning 0 for adSDKVersion.", versionString);
-  }
-  return version;
++ (nullable Class<GADAdNetworkExtras>)networkExtrasClass
+{
```

Comment:
Nit: Please remove newline between method name and the opening bracket.

The same applies to the other places.